src/refactor_nrepl/find/find_symbol.clj

```diff
@@ -228,20 +228,22 @@
                  (map (partial occurrence-for-optmap-default var-name)))]
         (sort-by :line-beg (concat local-occurrences optmap-def-occurrences))))))
 
-(defn- to-find-symbol-result
-  [{:keys [line-beg line-end col-beg col-end name file match]}]
-  [line-beg line-end col-beg col-end name file match])
-
 (defn find-symbol [{:keys [file ns name line column ignore-errors]}]
   (core/throw-unless-clj-file file)
   (let [macros (future (find-macro (core/fully-qualify ns name)))
         globals (->> (find-global-symbol file ns name (= ignore-errors "true"))
                      distinct
                      (remove find-util/spurious?)
                      future)]
+
     (or
      ;; find-local-symbol is the fastest of the three
-     (not-empty (remove find-util/spurious? (distinct (find-local-symbol file name line column))))
+     ;; if result is not empty, there is no point in keeping `find-macro` and `find-global-symbol` futures still active
+     (when-let [result (not-empty (remove find-util/spurious? (distinct (find-local-symbol file name line column))))]
+       (future-cancel macros)
+       (future-cancel globals)
```

Comment:
having found local-symbol we may cancel previously run (and potentially CPU intensive) futures looking for macros and global symbols?